### PR TITLE
[Release-1.5] Fix memory leak in TCP

### PR DESCRIPTION
**Pull request by gargnupur**
Signed-off-by: gargnupur <gargnupur@google.com>

Verified Envoy build's fine.

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Description: Fix memory leak in TCP
 Ref: https://github.com/istio/istio/issues/22007
Risk Level:
Testing:
Docs Changes:
Release Notes:
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: source/extensions/common/wasm/context.cc [hunk 1]
```diff
@@ -270,6 +270,16 @@ WasmResult Context::setTickPeriod(std::chrono::milliseconds tick_period) {
   return WasmResult::Ok;
 }
 
+void Context::onCloseTCP(){
+  if(tcp_connection_closed_){
```
Comment (PiotrSikora) [inline source/extensions/common/wasm/context.cc]: Please fix style (also, it looks like this is not cherry-pick of correct commit)?

**Comment by mandarjog** [comment]
@PiotrSikora do you know why this was not caught in asan? More tests ?

File: source/extensions/common/wasm/context.cc (hunk 1 above)
Comment (gargnupur) [reply source/extensions/common/wasm/context.cc]: Thanks for catching. Fixed format.

The other commit e64efd960cd8e97287764b9bae39b3a0def95046 is a merge and hence cherry-pick was not picking it.

**Comment by gargnupur** [comment]
@jplevyak , @PiotrSikora : closing this pr.. as it was not targeting release-1.5 :(

**Comment by gargnupur** [comment]
Created https://github.com/istio/envoy/pull/187 actually in release-1.5

**Pull request closed without merging**